scripts/svm/bridgeLiabilityToHubPool.ts

```diff
@@ -0,0 +1,305 @@
+
+const messageTransmitterAbi = [
+  {
+    inputs: [
+      {
+        internalType: "bytes",
+        name: "message",
+        type: "bytes",
+      },
+      {
+        internalType: "bytes",
+        name: "attestation",
+        type: "bytes",
+      },
+    ],
+    name: "receiveMessage",
+    outputs: [
+      {
+        internalType: "bool",
+        name: "success",
+        type: "bool",
+      },
+    ],
+    stateMutability: "nonpayable",
+    type: "function",
+  },
+  {
+    inputs: [
+      {
+        internalType: "bytes32",
+        name: "",
+        type: "bytes32",
+      },
+    ],
+    name: "usedNonces",
+    outputs: [
+      {
+        internalType: "uint256",
+        name: "",
+        type: "uint256",
+      },
+    ],
+    stateMutability: "view",
+    type: "function",
+  },
+];
+
+async function bridgeLiabilityToHubPool(): Promise<void> {
+  const seed = SOLANA_SPOKE_STATE_SEED; // Seed is always 0 for the state account PDA in public networks.
+
+  // Resolve Solana cluster, EVM chain ID, Iris API URL and USDC addresses.
+  let isDevnet: boolean;
+  const solanaRpcEndpoint = provider.connection.rpcEndpoint;
+  if (solanaRpcEndpoint.includes("devnet")) isDevnet = true;
+  else if (solanaRpcEndpoint.includes("mainnet")) isDevnet = false;
+  else throw new Error(`Unsupported solanaCluster endpoint: ${solanaRpcEndpoint}`);
+
+  const svmUsdc = isDevnet ? SOLANA_USDC_DEVNET : SOLANA_USDC_MAINNET;
+
+  const [statePda, _] = PublicKey.findProgramAddressSync(
+    [Buffer.from("state"), seed.toArrayLike(Buffer, "le", 8)],
+    svmSpokeProgram.programId
+  );
+
+  const irisApiUrl = isDevnet ? CIRCLE_IRIS_API_URL_DEVNET : CIRCLE_IRIS_API_URL_MAINNET;
+
+  const cctpMessageTransmitter = isDevnet
+    ? SEPOLIA_CCTP_MESSAGE_TRANSMITTER_ADDRESS
+    : MAINNET_CCTP_MESSAGE_TRANSMITTER_ADDRESS;
+
+  const messageTransmitter = new ethers.Contract(cctpMessageTransmitter, messageTransmitterAbi, ethersSigner);
+
+  const evmChainId = (await ethersProvider.getNetwork()).chainId;
+  const usdcAddress = TOKEN_SYMBOLS_MAP.USDC.addresses[evmChainId];
+  const usdc = BondToken__factory.connect(usdcAddress, ethersProvider);
+  const usdcBalanceBefore = await usdc.balanceOf(hubPoolAddress);
+
+  console.log("Receiving liability from Solana Spoke Pool to Ethereum Hub Pool...");
+  console.table([
+    { Property: "isTestnet", Value: process.env.TESTNET === "true" },
+    { Property: "hubPoolAddress", Value: hubPoolAddress },
+    { Property: "svmSpokeProgramProgramId", Value: svmSpokeProgram.programId.toString() },
+    { Property: "providerPublicKey", Value: provider.wallet.publicKey.toString() },
+    { Property: "usdcBalanceBefore", Value: usdcBalanceBefore.toString() },
+  ]);
+
+  const [transferLiability] = PublicKey.findProgramAddressSync(
+    [Buffer.from("transfer_liability"), new PublicKey(svmUsdc).toBuffer()],
```

Comment:
only comment I could make on this script (but I think it's fine as we do similar things all over the place) is this assumes we are working with USDC.

Reply:
Yeah, in many places we have dependencies on USDC, including in our smart contracts (solana adapter) and svm programs. Do you think it’s worth generalizing?